Added timestamp calculation to detsim city

MCruces-fz opened the pull request:
Timestamp calculation is based on `rate` parameter and `event number`.

MCruces-fz committed:
Modify output filename of all tests in detsim_test

gondiaz commented on invisible_cities/detsim/sensor_utils.py:
```diff
@@ -14,6 +15,27 @@
+    if rate == 0:
+        warnings.warn("Zero rate is unphysical, using default "
+                      "rate of 0.5 Hz instead")
+        return 0.5 / units.ms
```
The function output and the warning are inconsistent. Is it 0.5 Hz or 0.5 kHz? Remember that `1 Hz = 1/s`.

gondiaz commented on invisible_cities/detsim/sensor_utils.py:
```diff
@@ -14,6 +15,27 @@
+    -------
+    rate if rate != 0, else rate = 0.5
+    """
+    if rate == 0:
```
What if the rate is negative?

gondiaz commented on invisible_cities/detsim/sensor_utils_test.py:
```diff
@@ -115,3 +116,15 @@ def test_create_timestamp_greater_with_greater_eventnumber():
+@mark.filterwarnings("ignore:Zero rate")
+def test_get_rate_safe_allways_possitive():
+    """
+    Be sure that it woll always return the rate value such that
```
Typo (woll -> will). I would rewritte the docstring to something like "Check the rate is always positive/physical".

On the other hand, since you are testing a single value whose output is determined (hardcoded in the function being tested) I think it is better to check the exact result.

MCruces-fz committed:
Modify create_timestamp function

Added a check for physical rate values and keept the timestamp
calculation nested.

MCruces-fz committed:
Modify buffy city and source with new create timestamp function

MCruces-fz committed:
Modify detsim tests for new create_timestamp function

Update output files of test_detsim_exact and test_buffy_exact_result

MCruces-fz committed:
Modify detsim city and source with new create_timestamp function

  - Add rate variable (0.5 Hz) to detsim.conf
  - Add rate parameter in detsim function and MC_hits_from_files call

gondiaz commented on invisible_cities/cities/components.py:
```diff
@@ -114,6 +113,52 @@ def proxy(**kwds):
+                      "rate = 0.5 Hz instead")
+        rate = 0.5 * units.hertz
+    elif rate < 0:
+        warnings.warn(f"Negative rate is unphysical, using "
```
This warning outputs the rate in the default units of IC for 1/time. For example, try to see the warning comment for `rate = -1 * units.hertz`. It does not return the expected warning with positive 1 Hz.

MCruces-fz replied:
Ah, it is true. I should use `abs(rate) / units.hertz`, thanks

MCruces-fz committed:
Move create_timestamp function to components.py

MCruces-fz committed:
Add test for create_timestamp function

gondiaz reviewed (approved):
This PR succesfully implements timestamps for MC events in both detsim and buffy cities. Good job!
You can proceed with merging @carmenromo @halmamol @MiryamMV.
